dotnet/src/Connectors/Connectors.AI.OpenAI/KernelConfigOpenAIExtensions.cs

```diff
@@ -393,11 +395,27 @@ public static class KernelConfigOpenAIExtensions
 
     #endregion
 
-    private static HttpClient CreateHttpClient(this IDelegatingHandlerFactory handlerFactory,
-        ILogger? logger)
+    /// <summary>
+    /// Retrieves an instance of HttpClient.
+    /// </summary>
+    /// <param name="kernel">The kernel.</param>
+    /// <param name="httpClient">An optional pre-existing instance of HttpClient.</param>
+    /// <param name="logger">An optional logger.</param>
+    /// <returns>An instance of HttpClient.</returns>
+    private static HttpClient GetHttpClient(IKernel kernel, HttpClient? httpClient, ILogger? logger)
     {
-        var retryHandler = handlerFactory.Create(logger);
-        retryHandler.InnerHandler = new HttpClientHandler { CheckCertificateRevocationList = true };
-        return new HttpClient(retryHandler);
+        if (httpClient == null)
+        {
+            var retryHandler = kernel.Config.HttpHandlerFactory.Create(logger ?? kernel.Log);
+            retryHandler.InnerHandler = DefaultHttpClientHandler;
+            return new HttpClient(retryHandler, false); //We should refrain from disposing the underlying SK default HttpClient handler as it would impact other HTTP clients that utilize the same handler.
```

Comment:
As discussed offline, neither (1) or (2) matters because at this stage (.net runtime is in the process of cleaning up application resources and disposing the instance of the HttpClientHandler) all the requests should be completed and those that are not completed (app is closing after an HTTP request was sent but before the response was received) could not relevant anymore because application is closing/exiting.